src/components/Hoverable/index.js

```diff
@@ -102,7 +124,7 @@ class Hoverable extends Component {
             >
                 { // If this.props.children is a function, call it to provide the hover state to the children.
                     _.isFunction(this.props.children)
-                        ? this.props.children(this.state.isHovered)
+                        ? this.props.children(hovered)
```

Comment:
>Here. Hovered Flag depends on the scrolling as well. But if we are using onHoverIn or onHoverOut instead of this, they does not take scrolling in consideration. Line 58.

Sorry, I am still not seeing why this is relevant. Can you walk me through it? What is `onHoverOut` and `onHoverIn` used for?

>Next question is that adding scroll listener creates a small lag while scrolling. Because IMO, scrolling now re-render each rendered Hoverable item as soon as user starts scrolling. There could be many such nodes on the dom/vDom.

Why are they re-rendering?

Reply:
Could try delegating the events as you're adding a lot of listeners here as well we only need one to tell us if we are scrolling. If the scroll state isn't changing then there's no need to re-render the child. Think about it for a bit.

Reply:
A user can either use functional child and use hovered parameter to know the hovered state or he can use the callback props `onHoverIn` and `onHoverOut` to execute some code when user hover over and out. 

The `hovered` parameter which is passed to the functional children depends on scrolling as well But `onHoverIn` and `onHoverOut` callbacks are fired regardless of the scrolling state. Hope this is clear now. 

Why are they re-rendering?

because we update the state on start of the scrolling and end of it. State change should rerender the `Hoverable` components. As the scroll listeners are attached to each Hoverable. All of them we be rerendered. 

---

I can move out the scroll listener to a global lib but we still need to update the state on the Hoverable. So better would be use context-provider. Other than this, my old implementation was not utilizing the states. I'll think more about it but any directions will be helpful.

Reply:
>A user can either use functional child and use hovered parameter to know the hovered state or he can use the callback props onHoverIn and onHoverOut to execute some code when user hover over and out.

Do we care about these other examples? We're trying to fix an issue with the report page. Does the behavior break in cases where we use `onHoverIn` or `onHoverOut`?

>because we update the state on start of the scrolling and end of it. State change should rerender the Hoverable components. As the scroll listeners are attached to each Hoverable. All of them we be rerendered.

If the state change is the same there's no need to re-render the child. Can we optimize the rendering with `PureComponent`, `shouldComponentUpdate()` or `React.memo()`?

>I can move out the scroll listener to a global lib but we still need to update the state on the Hoverable. So better would be use context-provider. Other than this, my old implementation was not utilizing the states. I'll think more about it but any directions will be helpful.

Let's pause this part of the conversation until we confirm what the problem is.

Reply:
> Does the behavior break in cases where we use onHoverIn or onHoverOut?

I guess not. I will double check it. So it means we don't have worry about them.

> Can we optimize the rendering with PureComponent, shouldComponentUpdate() or React.memo()?

I will try this.